eng/bomgenerator/src/main/java/com/azure/tools/bomgenerator/DependencyAnalyzer.java

```diff
@@ -0,0 +1,238 @@
+                }
+            }
+            catch(Exception ex) {
+                System.out.println(ex);
+            }
+        }
+    }
+
+    private static MavenResolverSystemBase<PomEquippedResolveStage, PomlessResolveStage, MavenStrategyStage, MavenFormatStage> getMavenResolver() {
+        return Maven.configureResolver().withMavenCentralRepo(true);
+    }
+
+    private static void addDependencyToDependencyTree(BomDependency dependency, BomDependency parentDependency, Map<BomDependencyNoVersion, HashMap<String, Collection<BomDependency>>> dependencyTree) {
+        if (!dependencyTree.containsKey(dependency)) {
+            dependencyTree.put(new BomDependencyNoVersion(dependency.getGroupId(), dependency.getArtifactId()), new HashMap<>());
+        }
+
+        HashMap<String, Collection<BomDependency>> versionToParents = dependencyTree.get(dependency);
+        if(!versionToParents.containsKey(dependency.getVersion())) {
+            versionToParents.put(dependency.getVersion(), new ArrayList<>());
+        }
+
+        if(parentDependency != null) {
+            versionToParents.get(dependency.getVersion()).add(parentDependency);
+        }
+    }
+
+
+    private void makeDependencyInEligible(BomDependency dependency, String dropDependencyReason) {
+        if(nameToVersionToChildrenDependencyTree.containsKey(dependency)) {
+            HashMap<String, Collection<BomDependency>> versionToDependency = nameToVersionToChildrenDependencyTree.get(dependency);
+            bomIneligibleDependencies.put(dependency, dropDependencyReason);
+            if(dropDependencyReason == null) {
+                dropDependencyReason = dependency.toString();
+                logger.trace("\t\tDropping dependency {}", dependency.toString());
+            }
+            else {
+                logger.trace("\t\tDropping dependency {} due to dependency {}", dependency.toString(), dropDependencyReason);
+            }
+
+            // Make all the dependencies that include these also ineligible.
+            String finalDropDependencyReason = dropDependencyReason;
+            versionToDependency.get(dependency.getVersion()).forEach(parent -> makeDependencyInEligible(parent, finalDropDependencyReason));
+        }
+    }
+
+    private void pickCoreDependencyRoots() {
+        BomDependency coreDependency = inputDependencies.stream().filter(dependency -> dependency.getArtifactId().equals("azure-core")).findFirst().get();
+        coreDependencies.put(BomDependency.convertTo(coreDependency), coreDependency);
+        coreDependencies.putAll(getDependencies(coreDependency).stream().collect(Collectors.toMap(dependency -> BomDependency.convertTo(dependency), dependency -> dependency)));
+        this.bomEligibleDependencies.addAll(coreDependencies.values().stream().filter(dependency -> !externalDependencies.contains(dependency)).collect(Collectors.toList()));
```

Comment:
This may be a spot to use loops instead of streams to make the logic more clear and easier to step through